server/src/middleware/error-handler.ts

```diff
@@ -0,0 +1,23 @@
+import { NextFunction, Request, Response } from 'express';
+import { PortalErrors } from '../enum/errors';
+import { PortalErrorData } from '../interfaces/error-handler-interface';
+
+
+export class customError extends Error {
+    constructor(public errorEnumMessage: PortalErrors, public statusCode: number) {
+        super();
+    }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export function errorHandler(err: customError, req: Request, res: Response, next: NextFunction): void {
+    const error: PortalErrorData = {
+        message: err.errorEnumMessage.toString() || 'Es ist ein Fehler aufgetreten',
+        status: err.statusCode || 500,
+        type: PortalErrors[ err.errorEnumMessage ].toString(),
+        stacktrace: err.stack?.split('\n    '),
+        success: false,
+    };
+
+    res.status(error.status).send(error);
```

Comment:
zwei Sachen:
Wir verwenden, wenn wir Responses senden, immer den Wrapper, den Tomas geschrieben hat (wrapResponse).
Den bitte hier auch anwenden. Müsste dann so aussehen:
```suggestion
    const error: PortalErrorData = {
        message: err.errorEnumMessage.toString() || 'Es ist ein Fehler aufgetreten',
        status: err.statusCode || 500,
        type: PortalErrors[ err.errorEnumMessage ].toString(),
        stacktrace: err.stack?.split('\n    '),
        success: false,
    };

    res.status(error.status).send(wrapResponse(false, error));
```
Und dann noch was:
stacktrace generell komplett rausgeben finde ich nicht gut. Geh da bitte über ne Variable (Vars.config.logging = true/false), die das dann ausgibt oder eben nicht.

Reply:
Kann so nicht commited werden!
-> Import fehlt ;)

Reply:
den wrapResponse habe ich hier mit abgebildet... success: false --> der wrapResponse wird mit dem error-handler hinfällig

Reply:
prinzipiell ja, aber du hast nicht das Format:
{
  success: ..., 
  data: {
    ...
  }
}
Und auf dem Format arbeitet Tomas halt immer im Frontend. Also das wird erwartet. 
Deine Daten stehen immer in res.data. ...

Reply:
ok - ist implementiert - bitte testen ob es jetzt passt.